src/types/sequence.rs

```diff
@@ -257,6 +257,41 @@ impl PySequence {
     }
 }
 
+#[cfg(feature = "const-generics")]
+impl<'a, T, const N: usize> FromPyObject<'a> for [T; N]
+where
+    T: FromPyObject<'a>,
+{
+    #[cfg(not(feature = "nightly"))]
+    fn extract(obj: &'a PyAny) -> PyResult<Self> {
+        create_array_from_obj(obj)
+    }
+
+    #[cfg(feature = "nightly")]
+    default fn extract(obj: &'a PyAny) -> PyResult<Self> {
+        create_array_from_obj(obj)
+    }
+}
+
+#[cfg(all(feature = "const-generics", feature = "nightly"))]
+impl<'source, T, const N: usize> FromPyObject<'source> for [T; N]
+where
+    for<'a> T: FromPyObject<'a> + crate::buffer::Element,
+{
+    fn extract(obj: &'source PyAny) -> PyResult<Self> {
+        let mut array = create_array_from_obj(obj)?;
+        if let Ok(buf) = crate::buffer::PyBuffer::get(obj) {
+            if buf.dimensions() == 1 && buf.copy_to_slice(obj.py(), &mut array).is_ok() {
+                buf.release(obj.py());
+                return Ok(array);
+            }
+            buf.release(obj.py());
+        }
+        Ok(array)
```

Comment:
This is kind of tricky. `copy_to_slice` expects a slice of initialized memory and that is why I inverted the order of the operations. I.e.: From `copy_to_slice` -> `create_array_from_obj` to `create_array_from_obj` -> `copy_to_slice`

Maybe this can be improved using the `get_item` method of Buffer.